docs/concepts/fees.md

```diff
@@ -109,3 +109,278 @@ All data that is in an account's storage counts towards storage used. Even when
+
+Inevitably, network surges will cause higher fees. Users who might want to submit a transaction while the network usage is surging should consider sending the transaction at a later time to reduce costs.
+
+**Explain that the wallet might not allow the transaction due to a lack of funds**
+
+If dynamic fees increase to the highest possible level, the user’s fund might not be enough to execute the transaction. Let the users know that they should either add funds or try when the network is less busy.
+
+## How to learn more
+
+There are several places to learn more about transaction fees:
+
+- [FLIP-660](https://github.com/onflow/flow/pull/660)
+- [FLIP-753](https://github.com/onflow/flow/pull/753)
+- [Flow Fees Contract](https://github.com/onflow/flow-core-contracts/blob/master/contracts/FlowFees.cdc)
+
+> **Note**: If you have thoughts on the implementation of transaction fees on Flow, you can [leave feedback on this forum post](https://forum.onflow.org/t/variable-transaction-fees-are-coming-to-flow/2941).
+
+## FAQs
+
+**When will the fee update go into effect?**
+
+The updates were rolled out with the [Spork on April 6, 2022](../building-on-flow/nodes/node-operation/past-sporks.mdx#mainnet-17), and were enabled on [June 1st](https://forum.onflow.org/t/permissionless-contract-deployment-progress/2981) during the [weekly epoch transition](https://github.com/onflow/service-account/tree/main/transactions/set-execution-effort-weights/2022/jun-1).
+
+**Why are fees collected even when transactions fail?**
+
+Broadcasting and verifying a transaction requires execution, so costs are deducted appropriately.
+
+**What execution costs are considered above average?**
+
+There is no average for execution costs. Every function will vary significantly based on the logic implemented. You should review the optimization best practices to determine if you could reduce your costs.
+
+**Do hardware wallets like Ledger support segmented fees?**
+
+Yes.
+
+**What is the lowest execution cost?**
+
+The lowest execution cost is 1. This means your transaction included one function call or loop that didn't read or write any date.
+
+**Can I determine how much a transaction will cost on mainnet without actually paying?**
+
+You can estimate the costs in a two-way process: 1) determine execution costs for transactions (emulator or testnet) and 2) use an FCL SDK method to calculate the final transaction fees.
+
+**How accurate will testnet fees be to mainnet fees?**
+
+Final fees are determined by the surge factor on the network. The surge factor for the testnet will be different from the factor for the mainnet, so you need to expect a variation between mainnet and testnet estimates.
+
+**I use Blocto and I haven't paid any fees yet. Why is that?**
+
+That is because Blocto is acting as the payer for transactions. self-custody wallets may have the user pay the transaction. Additionally, apps can sponsor the transaction if they choose.
```

Comment:
```suggestion
That is because Blocto is acting as the payer for transactions. Self-custody wallets may have the user pay the transaction. Additionally, apps can sponsor the transaction if they choose.
```